lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -15,241 +15,188 @@
 
 tar_file = "test.tar.xz"
 tmp_dir = pathlib.Path("nonexistent/tmp/dir")
+tm_params = {
+    "benchmark_run_dir": "",
+    "channel_prefix": "",
+    "tds_hostname": "test.hostname.com",
+    "tds_port": 4242,
+    "controller": "test.hostname.com",
+    "group": "",
+    "hostname": "test.hostname.com",
+    "label": None,
+    "tool_metadata": {"persistent": {}, "transient": {}},
+    "tools": [],
+}
+
+
+@pytest.fixture
+def tool_meister():
+    return ToolMeister(
+        pbench_install_dir=None,
+        tmp_dir=None,
+        tar_path="tar_path",
+        sysinfo_dump=None,
+        params=tm_params,
+        redis_server=None,
+        logger=logging.getLogger(),
+    )
 
 
-@pytest.fixture()
-def mock_tar(monkeypatch):
-    def fake_run(*args, **kwargs):
-        assert kwargs["cwd"] == tmp_dir.parent
-        assert kwargs["stdin"] is None
-        assert kwargs["stderr"] == subprocess.STDOUT
-        assert kwargs["stdout"] == subprocess.PIPE
-        c = subprocess.CompletedProcess(args, returncode=0, stdout=b"", stderr=None)
-        assert all(
-            x in c.args[0]
-            for x in [
-                "tar_path",
-                "--create",
-                "--xz",
-                "--force-local",
-                f"--file={tar_file}",
-            ]
-        )
-        return c
-
-    monkeypatch.setattr(subprocess, "run", fake_run)
-
-
-@pytest.fixture(scope="function")
-def logger_fixture(request):
-    marker = request.node.get_closest_marker("log_level")
-
-    # Create a logger using a test function name, that called this fixture
-    logger = logging.getLogger(f"{request.function.__name__ }")
-
-    log_level = marker.args[0] if marker else logging.ERROR
-    logger.setLevel(log_level)
-    yield logger
-    logger.setLevel(level=logging.NOTSET)
+class TestCreateTar:
+    """Test the ToolMeister._create_tar() method behaviors."""
 
+    @staticmethod
+    def test_create_tar(tool_meister, monkeypatch):
```

Comment:
Why `staticmethod` though?

Reply:
There is no need for a `self` parameter for these tests.